code/modules/vehicles/ridden.dm

```diff
@@ -42,7 +42,7 @@
 /obj/vehicle/ridden/AltClick(mob/user)
 	if(!inserted_key || (!user.Adjacent(src) && user.dextrous))
 		return ..()
-	if(!is_occupant(user))
+	if(!is_occupant(user) && occupants)
```

Comment:
?

Reply:
I added it so you can remove the key without standing on the bike, if no one is riding it.